pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
(in otherwords: creating a flag who's value is directly correlated to the image name seems unnecessary.  that flag would basically be named "isScratchImage", given how it would be used at this point)